ibis/pandas/tests/test_operations.py

```diff
@@ -1193,3 +1189,247 @@ def test_quantile_array_access(client, t, df):
+
+
+@pytest.mark.xfail(AssertionError, reason='NYI')
+def test_batting_most_hits(players, players_df):
+    expr = players.mutate(
+        hits_rank=lambda t: t.H.rank().over(
+            ibis.window(order_by=ibis.desc(t.H))
+        )
+    )
+    result = expr.execute()
+    hits_rank = players_df.groupby('playerID').H.rank(
+        method='min', ascending=False
+    )
+    expected = players_df.assign(hits_rank=hits_rank)
+    tm.assert_frame_equal(result[expected.columns], expected)
+
+
+def test_batting_quantile(players, players_df):
+    expr = players.mutate(hits_quantile=lambda t: t.H.quantile(0.25))
+    result = expr.execute()
+    hits_quantile = players_df.groupby('playerID').H.transform(
+        'quantile', 0.25
+    )
+    expected = players_df.assign(hits_quantile=hits_quantile)
+    tm.assert_frame_equal(result[expected.columns], expected)
+
+
+@pytest.mark.parametrize('op', ['sum', 'mean', 'min', 'max'])
+def test_batting_specific_cumulative(batting, batting_df, op):
+    ibis_method = methodcaller('cum{}'.format(op))
+    expr = ibis_method(batting.sort_by([batting.yearID]).G)
+    result = expr.execute().astype('float64')
+
+    pandas_method = methodcaller(op)
+    expected = pandas_method(
+        batting_df[['G', 'yearID']].sort_values('yearID').G.expanding()
+    ).reset_index(drop=True)
+    tm.assert_series_equal(result, expected)
+
+
+def test_batting_cumulative(batting, batting_df):
+    expr = batting.mutate(
+        more_values=lambda t: t.G.sum().over(ibis.window(order_by=t.yearID))
+    )
+    result = expr.execute()
+
+    columns = ['G', 'yearID']
+    more_values = batting_df[columns].sort_values('yearID').G.cumsum()
+    expected = batting_df.assign(more_values=more_values)
+
+    tm.assert_frame_equal(result[expected.columns], expected)
+
+
+def test_batting_cumulative_partitioned(batting, batting_df):
+    expr = batting.mutate(
+        more_values=lambda t: t.G.sum().over(ibis.window(
+            order_by=t.yearID,
+            group_by=t.lgID,
+        ))
+    )
+    result = expr.execute().more_values
+
+    columns = ['G', 'yearID', 'lgID']
+    key = 'lgID'
+    expected_result = batting_df[columns].groupby(
+        key, sort=False, as_index=False
+    ).apply(lambda df: df.sort_values('yearID')).groupby(
+        key, sort=False
+    ).G.cumsum().sort_index(level=-1)
+    expected = expected_result.reset_index(
+        list(range(expected_result.index.nlevels - 1)),
+        drop=True
+    ).reindex(batting_df.index)
+    expected.name = result.name
+
+    tm.assert_series_equal(result, expected)
+
+
+def test_batting_rolling(batting, batting_df):
+    expr = batting.mutate(
+        more_values=lambda t: t.G.sum().over(
+            ibis.window(order_by=t.yearID, preceding=5)
+        )
+    )
+    result = expr.execute()
+
+    columns = ['G', 'yearID']
+    more_values = batting_df[columns].sort_values('yearID').G.rolling(5).sum()
```

Comment:
Do you think this will be able to easily handle temporal expressions like "5 days" here?